includes/base/controls-stack.php

```diff
@@ -431,9 +437,41 @@ public function add_control( $id, array $args, $options = [] ) {
 			}
 		}
 
+		if ( $should_optimize_controls ) {
+			unset(
+				$args['label_block'],
+				$args['label'],
+				$args['tab'],
+				$args['options'],
+				$args['separator'],
+				$args['size_units'],
+				$args['range'],
+				$args['render_type'],
+				$args['toggle'],
+				$args['ai'],
+				$args['label_on'],
+				$args['label_off'],
+				$args['labels'],
+				$args['handles'],
+			);
+		}
+
 		return Plugin::$instance->controls_manager->add_control_to_stack( $this, $id, $args, $options );
 	}
 
+	private function should_optimize_controls() {
+		static $retval = null;
+
+		if ( null === $retval ) {
+			$retval = (
+				! is_admin()
+				&& ! \Elementor\Plugin::$instance->preview->is_preview_mode()
+			);
+		}
+
+		return $retval;
+	}
```

Comment:
Is this basically just a check for the front end? Do we need to consider the following as well?

`( ! is_admin() || wp_doing_ajax() ) && ! wp_doing_cron()` <-- Not sure if these actions will invoke this file or not, but in general this is a good WordPress way to be sure we are on Frontend.

```
$retval = (
	! is_admin()
	&& ! \Elementor\Plugin::$instance->preview->is_preview_mode()
);
```
Maybe this logic can be moved to a helper like `\Elementor\Plugin::$instance->frontend->is_frontend()`? I also need to do FE checks in another place so it might be helpful to have one source.

Reply:
This is the first draft of it. Right now, I make it on the front end only, and I'm not sure if we can expand it to other areas.